sheets/api.py

```diff
@@ -253,31 +261,111 @@ def batch_update_sheet_cells(self, sheet_id, request_objects):
             .execute()
         )
 
-    def request_file_watch(self, file_id, channel_id):
-        """
-        Executes the request to watch for changes made to a specific file in Drive. If successful, Drive will
-        make requests to our webhook when changes are made to the given file.
 
-        Returns:
-            dict: Google API response to a files.watch request
-        """
-        return (
-            self.pygsheets_client.drive.service.files()
-            .watch(
-                fileId=file_id,
-                body={
-                    "id": channel_id,
-                    "address": urljoin(
-                        settings.SITE_BASE_URL,
-                        reverse("handle-coupon-request-sheet-update"),
-                    ),
-                    "payload": True,
-                    "kind": GOOGLE_API_FILE_WATCH_KIND,
-                    "type": GOOGLE_API_NOTIFICATION_TYPE,
-                },
-            )
-            .execute()
+def request_file_watch(file_id, channel_id, expiration=None, credentials=None):
+    """
+    Sends a request to the Google API to watch for changes in a given file. If successful, this
+    app will receive requests from Google when changes are made to the file.
+    Ref: https://developers.google.com/drive/api/v3/reference/files/watch
+
+    Args:
+        file_id (str): The id of the file in Google Drive (can be determined from the URL)
+        channel_id (str): Arbitrary string to identify the file watch being set up. This will
+            be included in the header of every request Google sends to the app.
+        expiration (datetime.datetime or None): The datetime that this file watch should expire.
+            Defaults to 1 hour, and cannot exceed 24 hours.
+        credentials (google.oauth2.credentials.Credentials or None): Credentials to be used by the
+            Google Drive client
+
+    Returns:
+        dict: The Google file watch API response
+    """
+    credentials = credentials or get_credentials()
+    # Including cache_discovery=False to suppress a noisy warning
+    drive_service = build("drive", "v3", credentials=credentials, cache_discovery=False)
+    extra_body_params = {}
+    if expiration:
+        extra_body_params["expiration"] = format_datetime_for_google_timestamp(
+            expiration
+        )
+    return (
+        drive_service.files()
+        .watch(
+            fileId=file_id,
+            supportsTeamDrives=True,
+            body={
+                "id": channel_id,
+                "resourceId": file_id,
+                "address": urljoin(
+                    settings.SITE_BASE_URL,
+                    reverse("handle-coupon-request-sheet-update"),
+                ),
+                "payload": True,
+                "kind": GOOGLE_API_FILE_WATCH_KIND,
+                "type": GOOGLE_API_NOTIFICATION_TYPE,
+                "expiration": "failfailfail"
+                # **extra_body_params,
```

Comment:
Should `"expiration": "failfailfail"` be removed and `**extra_body_params` uncommented?

Reply:
😬